### ISSUE TITLE
Build fails with -DBUILD_SHARED_LIBS=On

### ISSUE THREAD

**Issue by Sameeranjoshi** [facebookincubator/BOLT#142]
Hello,
I am trying to build llvm bolt using the mono repo(latest llvm-project HEAD).
I don't see the specific commit which you are checking out to.
```
fatal: bad object f137ed238db11440f03083b1c88b7ffc0f4af65e
```
1. Does this mean that the commits are based out of the 
``` https://github.com/llvm-mirror/llvm llvm``` branch?

Are there any steps documented to build the tool?
Thank you!

**Comment by rafaelauler** [comment]
Hi @Sameeranjoshi 

Try the rebased branch https://github.com/facebookincubator/BOLT/tree/rebased

**Comment by Sameeranjoshi** [comment]
Thank you @rafaelauler the branch builds for me, except when adding below cmake invocations.

```
  -DLLVM_LIT_ARGS=-v \
  -DLLVM_CCACHE_BUILD=On \
  -DBUILD_SHARED_LIBS=On \
  -DLLVM_OPTIMIZED_TABLEGEN=On \
```
Do shared build work?

**Comment by rafaelauler** [comment]
Thanks for reporting, the build indeed is broken for shared libs. I was able to repro with

-DBUILD_SHARED_LIBS=On

**Comment by Sameeranjoshi** [comment]
Hi does this seem like a newbie issue to solve?
How much efforts are needed for this?  Are there any related references to look into(I think LLVM sources of some specific project which supports shared builds) ?
Given that I reported it would be good if I try solving the same.

Thank you.

**Comment by rafaelauler** [comment]
I took an initial look at that and it seems we may have a dependency problem that can only be solved by reorganizing our source code structure, so that might be too difficult of a task. But if you find a way to make it work in our CMakeFiles, this would be welcome (even if it involves disabling completely the shared lib build, but I wasn't able to do that in my initial investigation).

**Comment by aaupov** [comment]
Fixed with 43d78552e6a92042fc81416c5948f0b5b4fa8596
